### PR TITLE
LPD-40819 Removed Developer License Price for Cloud Apps (sorted)

### PR DESCRIPTION
**Pull request by CalebJamesH**
https://liferay.atlassian.net/jira/software/c/projects/LPD/boards/256?quickFilter=4481&selectedIssue=LPD-40819

### CODE REVIEW

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx [hunk 1]
```diff
@@ -87,27 +92,34 @@ export function ProvideVersionDetailsPage({
...
 		let newOptionId: number;
 
 		if (!optionId && !targetOption) {
 			newOptionId = await postOption(
-				isDXP ? getDxpOptionBody() : getTrialOptionBody()
+				isCloud
+					? getCloudOptionBody()
+					: isDXP
+						? getDxpOptionBody()
+						: getTrialOptionBody()
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx:111]:
```suggestion
				isCloud
					? getCloudOptionBody()
					: getDxpOptionBody()
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx [hunk 2]
```diff
@@ -87,27 +92,34 @@ export function ProvideVersionDetailsPage({
...
 		}
 
-		const productOption = isDXP
-			? getDxpProductOptionBody(newOptionId)
-			: getTrialProductOptionBody(newOptionId);
+		const productOption = isCloud
+			? getCloudProductOptionBody(newOptionId)
+			: isDXP
+				? getDxpProductOptionBody(newOptionId)
+				: getTrialProductOptionBody(newOptionId);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx:122]:
```suggestion
		const productOption = isCloud
			? getCloudProductOptionBody(newOptionId)
			: getDxpProductOptionBody(newOptionId)
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx [hunk 3]
```diff
@@ -124,17 +136,19 @@ export function ProvideVersionDetailsPage({
 			type: TYPES.UPDATE_PRODUCT_OPTION_ID,
 		});
 
-		if (isDXP) {
-			const [standardOptionId, developerOptionId, trialOptionId] =
+		if (isCloud || isDXP) {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx:139]: if (isCloud || isDXP) { remove

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-site-initializer/site-initializer/commerce-catalogs/catalog.options.json [hunk 4]
```diff
@@ -54,6 +54,30 @@
 			}
 		]
 	},
+	{
+		"ddmFormFieldTypeName": "radio",
+		"description": "Specifies if a acquision of an app.",
+		"externalReferenceCode": "MKT-OPTION-CLOUD-USAGE",
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-site-initializer/site-initializer/commerce-catalogs/catalog.options.json:60]:
```suggestion
		"externalReferenceCode": "MKT-OPTION-CLOUD-LICENSE-USAGE-TYPE",
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by CalebJamesH** [comment]
Just starting reviewing :)

**Review by kevenleone — APPROVED** [review]

**Comment by kevenleone** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
ci:stop

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**